tests/kernel/module_build.pm

```diff
@@ -0,0 +1,57 @@
+# SUSE's openQA tests
+#
+# Copyright © 2019 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Test kernel module build
+#          Example http://www.tldp.org/LDP/lkmpg/2.6/html/x121.html
+# Maintainer: Petr Cervinka <pcervinka@suse.com>
+# Tags: https://progress.opensuse.org/issues/49031
+
+use base 'consoletest';
+use strict;
+use warnings;
+use testapi;
+use utils;
+
+sub run {
+    my $self = shift;
+    select_console "root-console";
+    zypper_call "in kernel-default-devel";
+
+    my $module_src = <<'EOF';
+#include <linux/module.h>
+#include <linux/kernel.h>
+int init_module(void)
+{
+	printk(KERN_INFO "Hello world 1.]\n");
+	return 0;
+}
+void cleanup_module(void)
+{
+	printk(KERN_INFO "Goodbye world 1.\n");
+}
+EOF
+    my $makefile = << 'EOF';
+obj-m += hello.o
+all:
+\tmake -C /lib/modules/$(shell uname -r)/build M=$(PWD) modules
+clean:
+\tmake -C /lib/modules/$(shell uname -r)/build M=$(PWD) clean    
+EOF
+    assert_script_run "echo '$module_src' >> hello.c";
```

Comment:
looks fine from perl perspective , but really bad from openQA. Maybe you haven't realized that you actually **typing** the file word by word  in terminal ? I would go for file creation in data folder and downloading it as it is

Reply:
well, i wanted to avoid to use data files, but you are right

Reply:
moved to data/kernel/module